hawkular-alerts-api/src/main/java/org/hawkular/alerts/api/model/trigger/Trigger.java

```diff
@@ -28,69 +34,144 @@
  * @author Jay Shaughnessy
  * @author Lucas Ponce
  */
-public class Trigger extends TriggerTemplate {
+public class Trigger {
 
-    public enum Mode {
-        FIRING, AUTORESOLVE
-    };
+    @JsonInclude
+    private String tenantId;
 
+    /** Unique within the tenant */
     @JsonInclude
     private String id;
 
+    /** For display */
+    @JsonInclude
+    private String name;
+
+    @JsonInclude(Include.NON_EMPTY)
+    protected Map<String, String> context;
+
+    /** A map with key based on actionPlugin and value a set of action's ids */
+    @JsonInclude(Include.NON_EMPTY)
+    private Map<String, Set<String>> actions;
+
+    /** Disable automatically after firing */
+    @JsonInclude
+    private boolean autoDisable;
+
+    /** Enable automatically if disabled and resolved manually */
+    @JsonInclude
+    private boolean autoEnable;
+
+    /** Switch to auto-resolve mode after firing */
+    @JsonInclude
+    private boolean autoResolve;
+
+    /** Resolve all unresolved alerts when auto-resolve condition-set is satisfied */
+    @JsonInclude
+    private boolean autoResolveAlerts;
+
+    @JsonInclude
+    private Match autoResolveMatch;
+
+    /** Is a member trigger of the specified group triggerId */
+    @JsonInclude(Include.NON_EMPTY)
+    private String memberOf;
```

Comment:
You are correct:   if 'group' flag is true then 'memberOf' is irrelevant/null and 'orphan' is irrelevant/false.   if memberOf is set then this is a member trigger and 'group' is false.  A member trigger is a trigger evaluated like any other, but is managed via the group.  An orphan is member trigger that is no longer managed by, but is still assigned to the group.  I have added jdoc in Trigger class to help make this clear, on setGroup(), setMemberOf() and setOrphan().